R/surv.b.R

```diff
@@ -40,150 +52,237 @@ survClass <- R6::R6Class(
+                # ------- add the df and the p-values
+                DF$df <- df
+                DF$pvalue <- pchisq(DF$chisqr, df, lower.tail = FALSE)
                 
-                if (length(groups) >= 2) {
-                    groupsData <- list()
-                    for (group in groups) {
-                        ss <- (data$group == group)
-                        groupsData[[group]] <- subset(data, subset=ss, select=c('elapsed', 'event'))
-                    }
-                }
                 
-                for (pair in tt$rowKeys) {
-                    
-                    x <- groupsData[[pair[1]]]
-                    y <- groupsData[[pair[2]]]
+                #################################
+                ### Populates the table with the  
+                ### tests results
+                #################################
+                tt <- self$results$tests
+                if (tt$isNotFilled() && length(self$options$tests) > 0) {
                     
                     for (i in seq_along(self$options$tests)) {
+                        
                         test <- self$options$tests[i]
                         
-                        if (tt$isFilled(rowKey=pair, col=paste0('nu[', test, ']')))
+                        if (tt$isFilled(rowKey=1, col=paste0('chisqr[', test, ']')))
                             next()
                         
                         private$.checkpoint()
-                        
-                        result <- EnvStats::twoSampleLinearRankTestCensored(
-                            test = test,
-                            x = x$elapsed,
-                            x.censored = ! x$event,
-                            y = y$elapsed,
-                            y.censored = ! y$event,
-                            censoring.side = 'right')
-                        
                         row <- list()
+                        row[[paste0('chisqr[', test, ']')]] <- DF$chisqr[i]
+                        row[[paste0('df[', test, ']')]] <- DF$df[i]
+                        row[[paste0('pvalue[', test, ']')]] <- DF$pvalue[i]
                         
-                        row[[paste0('z[', test, ']')]] <- result$statistic['z']
-                        row[[paste0('nu[', test, ']')]] <- result$statistic['nu']
-                        row[[paste0('nuse[', test, ']')]] <- sqrt(result$statistic['var.nu'])
-                        row[[paste0('p[', test, ']')]] <- result$p.value
-                        
-                        tt$setRow(rowKey=pair, values=row)
+                        tt$setRow(rowKey=1, values=row)
                     }
+                    
                 }
+                tt$setNote("note1","Log-Rank and Peto-Peto tests use survdiff from the survival package. Gehan and Tarone-Ware tests are obtained from the coin package.",init=TRUE)
+
             }
+
+            
+            
+            
+
+            
```

Comment:
lot of space here

Reply:
Lol I tend to do that many times. It's like fresh air for me when programming :-) Too much air this time ! I've just removed it